localstack/services/stepfunctions/asl/parse/preprocessor.py

```diff
@@ -191,6 +191,10 @@
 
 
 class Preprocessor(ASLParserVisitor):
+    def __init__(self, account_id: str, region_name: str):
```

Comment:
Adding the values to the environment is a decent solution, and evaluation components can produce runtime values like region and account during evaluation. So I don't think we should pollute the Preprocessor with runtime values, as conceptually it just builds the evaluation tree.

Reply:
When the `Preprocessor` constructs the `Resource` object from ARN, the ARN does not necessarily contain the region and the account ID. So it is necessary that the correct account ID and region from the request context are passed down.

https://github.com/localstack/localstack/blob/aad9b80492fdc9f7b22f48af0184cb5febf6dd23/localstack/services/stepfunctions/asl/parse/preprocessor.py#L238

Using `aws_stack.get_region()` is not an option as it doesn't always have the proper region. Moreover, I'll be deprecating that method in the future for this exact reason.